manifests/forwarder.pp

```diff
@@ -75,26 +81,13 @@
     install_options => $splunk::params::forwarder_install_options,
     tag             => 'splunk_forwarder',
   }
+  Splunkforwarder_input{ tag => 'splunk_forwarder'}
+  Splunkforwarder_output{ tag => 'splunk_forwarder'}
 
   # Declare inputs and outputs specific to the forwarder profile
-  splunkforwarder_input { 'default_host':
-    section => 'default',
-    setting => 'host',
-    value   => $::clientcert,
-    tag     => 'splunk_forwarder',
-  }
-  splunkforwarder_output { 'tcpout_defaultgroup':
-    section => 'default',
-    setting => 'defaultGroup',
-    value   => "${server}_${logging_port}",
-    tag     => 'splunk_forwarder',
-  }
-  splunkforwarder_output { 'defaultgroup_server':
-    section => "tcpout:${server}_${logging_port}",
-    setting => 'server',
-    value   => "${server}:${logging_port}",
-    tag     => 'splunk_forwarder',
-  }
+   create_resources( 'splunkforwarder_input', hiera_hash('splunk::forwarder::forwarder_input',$::splunk::forwarder::forwarder_input))
```

Comment:
We would prefer that hiera function calls not happen inside public modules, as many users don't have hiera and this will cause errors. Could you add parameters to pass in the forwarder_input and output values so that users can pass in a hash directly, use the defaults as before, or call hiera_hash to pass in the hash?